app/uk/gov/hmrc/merchandiseinbaggagefrontend/forms/PurchaseDetailsFormProvider.scala

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.merchandiseinbaggagefrontend.forms
+
+import javax.inject.Inject
+import play.api.data.Form
+import play.api.data.Forms.mapping
+import uk.gov.hmrc.merchandiseinbaggagefrontend.forms.mappings.Mappings
+import uk.gov.hmrc.merchandiseinbaggagefrontend.model.core.PurchaseDetailsInput
+
+class PurchaseDetailsFormProvider @Inject() extends Mappings {
+
+  def apply(): Form[PurchaseDetailsInput] =
```

Comment:
Not necessary to do now, but the following also seems to work:

object PurchaseDetailsForm extends Mappings

and then just reference the form directly from the controller.

To be continued :-)

Reply:
Okay we can refactor these to be Objects rather than Classes as a bit of a tidy up exercise and then establish the pattern going forward